### DISCUSSION TITLE
Can`t visualize Green function

### DISCUSSION THREAD

**Discussion by DFTsher** [TRIQS/solid_dmft#77] [Q&A] [answered]
Hello triqs community!
I try to do some basic tutorial of triqs, such as [The Anderson impurity model](https://triqs.github.io/cthyb/latest/guide/aim.html) and [Multiorbital impurity model]https://triqs.github.io/cthyb/latest/guide/slater_five_band.html. And calculation goes well. But when I try to visualize the Green's function using the codes provided on the corresponding exercise pages, nothing happens (no massages, no new files etc). There is a feeling that triqs simply does not find or does not use triqs.gf. Can you help me with this issue?

**Comment by the-hampel** [comment]
Hi @DFTsher , 
let maybe stick to the first tutorial you are mentioning: https://triqs.github.io/cthyb/latest/guide/aim.html . Did the first part work, e.g. run the DMFT iterations and store results into the h5 archive: 
```
if mpi.is_master_node():
    with HDFArchive("aim_solution.h5",'w') as Results:
        Results["G_tau"] = S.G_tau
        Results["G_iw"] = S.G_iw
        Results["G_l"] = S.G_l
```
Then you should find a new file `aim_solution.h5` in your directory? And then you run: 
```
with HDFArchive('aim_solution.h5','r') as ar:
    oplot(ar['G_iw']['up'], '-o', x_window = (0,10))
```
and this does not work you say? What exactly is the error message given to you from python? 

I just tried this with the recent triqs/3.2.x version and it works as written in the tutorial.

**Reply by DFTsher** [reply]
The-hampel thanks for such a quick response.
Yes, after using trqis-cthyb solver I get this file, but I don`t know if it is correct or not. 
[aim_solution.zip](https://github.com/TRIQS/solid_dmft/files/14835655/aim_solution.zip)
Plus when I  do my calculation triqs gives me this message: Warning: could not identify MPI environment!
And what command did you use to start calculation?

**Reply by the-hampel** [reply]
Yes the results are correct. This is the output I get when running: 
```
with HDFArchive('aim_solution.h5','r') as ar:
    oplot(ar['G_iw']['up'], '-o', x_window = (0,10))
```
![image](https://github.com/TRIQS/solid_dmft/assets/21337117/bb8e7492-ca8b-45a4-a426-6d3a4b95e038)

So this is all good. `Warning: could not identify MPI environment!` is just a warning. You can ignore it. This is because you run triqs/cthyb only in jupyter / simple python without MPI. As long as you don't do longer computations this is totally fine. 

Now, what exactly is the error message you get when trying to plot the results?

**Reply by DFTsher** [reply]
Well, nothing, few seconds and empty command line. By exception method (the triqs.gf and h5 packages were used in the aim.py file), it could be that trix cannot find the triqs.plot.mpl_interface package. Or it could be a rendering issue on the cluster itself. If so, is it possible to save the graph to a file?

**Reply by the-hampel** [reply]
Where and how do you execute the script? The plotting is only possible if you are in an active X session. This is using matplotlib to show the plot. If you run this on the cluster via ssh likely nothing will happen and wait forever to close the plot. The visualization should be done afterwards locally.

**Reply by DFTsher** [reply]
I ran it on ssh cluster with x11. I guess that was a problem. Is there any instructions how to run it locally?

**Reply by the-hampel** [reply, marked as answer]
Running the solver etc. on the cluster is exactly how it is intended to be done. You can simply transfer the h5 file to your local laptop and then run the last plotting lines and all post-processing in a jupyter notebook on your laptop (or maybe even on the cluster if your cluster provides jupyterhub or similar)? If you need help installing triqs on your laptop you can find help here: https://triqs.github.io/triqs/latest/install.html .
